src/main/java/gray/Tasks.java

```diff
@@ -0,0 +1,91 @@
+package gray;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import gray.task.Task;
+
+/**
+ * Represents the tasks API
+ */
+public class Tasks {
+    private final File saveFile;
+    private final ArrayList<Task> tasks = new ArrayList<>();
+
+    /**
+     * Constructs a tasks object
+     * @param saveFile
+     */
+    public Tasks(File saveFile) {
+        this.saveFile = saveFile;
+        load();
+    }
+
+    private void save() {
+        saveFile.getParentFile().mkdirs();
+        try {
+            Utility.serialise(saveFile, tasks);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void load() {
+        saveFile.getParentFile().mkdirs();
+        if (!saveFile.exists()) {
+            return;
+        }
+        try {
+            ArrayList<Task> taskList = (ArrayList<Task>) Utility.deserialise(saveFile);
+            tasks.clear();
+            tasks.addAll(taskList);
+        } catch (IOException | ClassNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public int size() {
+        return tasks.size();
+    }
```

Comment:
Very very minor nitpick. I understand the use of `size()` as a natural continuation of `ArrayList::size`, since the tasks are stored in the class as an `ArrayList` anyway. But maybe naming the method differently, e.g. `tasks.count()` would be clearer than `tasks.size()`?